rocketmq-broker/src/test/java/com/alibaba/rocketmq/broker/api/SendMessageTest.java

```diff
@@ -20,68 +20,68 @@
 
-        brokerController.start();
+    MQClientAPIImpl client = new MQClientAPIImpl(new NettyClientConfig(), null, null, new ClientConfig());
+    String topic = "UnitTestTopic";
 
-        MQClientAPIImpl client = new MQClientAPIImpl(new NettyClientConfig(), null, null, null);
+    @Before
+    @Override
+    public void startup() throws Exception {
+        super.startup();
         client.start();
 
-        for (int i = 0; i < 100; i++) {
-            String topic = "UnitTestTopic_" + i % 3;
-            Message msg = new Message(topic, "TAG1 TAG2", "100200300", ("Hello, Nice world\t" + i).getBytes());
-            msg.setDelayTimeLevel(i % 3 + 1);
-
-            try {
-                SendMessageRequestHeader requestHeader = new SendMessageRequestHeader();
-                requestHeader.setProducerGroup("abc");
-                requestHeader.setTopic(msg.getTopic());
-                requestHeader.setDefaultTopic(MixAll.DEFAULT_TOPIC);
-                requestHeader.setDefaultTopicQueueNums(4);
-                requestHeader.setQueueId(i % 4);
-                requestHeader.setSysFlag(0);
-                requestHeader.setBornTimestamp(System.currentTimeMillis());
-                requestHeader.setFlag(msg.getFlag());
-                requestHeader.setProperties(MessageDecoder.messageProperties2String(msg.getProperties()));
-
-                SendResult result = client.sendMessage("127.0.0.1:10911", "brokerName", msg, requestHeader, 1000 * 5,
-                        CommunicationMode.SYNC, new SendMessageContext(), null);
-                System.out.println(i + "\t" + result);
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
-        }
+    }
 
+    @After
+    @Override
+    public void shutdown() throws Exception {
         client.shutdown();
+        super.shutdown();
+    }
 
-        brokerController.shutdown();
+    @Test
+    public void testSendSingle() throws Exception {
+        Message msg = new Message(topic, "TAG1 TAG2", "100200300", "body".getBytes());
+        try {
+            SendMessageRequestHeader requestHeader = new SendMessageRequestHeader();
+            requestHeader.setProducerGroup("abc");
+            requestHeader.setTopic(msg.getTopic());
+            requestHeader.setDefaultTopic(MixAll.DEFAULT_TOPIC);
+            requestHeader.setDefaultTopicQueueNums(4);
+            requestHeader.setQueueId(0);
+            requestHeader.setSysFlag(0);
+            requestHeader.setBornTimestamp(System.currentTimeMillis());
+            requestHeader.setFlag(msg.getFlag());
+            requestHeader.setProperties(MessageDecoder.messageProperties2String(msg.getProperties()));
+
+            SendResult result = client.sendMessage(brokerAddr, BROKER_NAME, msg, requestHeader, 1000 * 5,
+                    CommunicationMode.SYNC, new SendMessageContext(), null);
+            assertTrue(result.getSendStatus() == SendStatus.SEND_OK);
+        } catch (Exception e) {
```

Comment:
Good Job @WillemJiang  Later, we will draft some unit test best practice for community guys